tests/drivers/sensor/accel/src/main.c

```diff
@@ -55,27 +86,63 @@ static void run_tests_on_accel(const struct device *accel)
 ZTEST_USER_F(sensor_accel, test_sensor_accel_basic_spi)
 {
 	run_tests_on_accel(fixture->accel_spi);
-	test_sensor_accel_basic(fixture->accel_spi);
+	test_sensor_accel_basic(fixture->accel_spi, fixture->accel_emul_spi);
 }
 
-
 ZTEST_USER_F(sensor_accel, test_sensor_accel_basic_i2c)
 {
 	if (fixture->accel_i2c == NULL) {
 		ztest_test_skip();
 	}
 
 	run_tests_on_accel(fixture->accel_i2c);
-	test_sensor_accel_basic(fixture->accel_i2c);
+	test_sensor_accel_basic(fixture->accel_i2c, fixture->accel_emul_i2c);
+}
+
+static void sensor_accel_setup_emulator(const struct device *dev, const struct emul *accel_emul)
+{
+	if (accel_emul == NULL) {
+		return;
+	}
+
+	static struct {
+		enum sensor_channel channel;
+		q31_t value;
+	} values[] = {
+		{SENSOR_CHAN_ACCEL_X, 0},       {SENSOR_CHAN_ACCEL_Y, 1 << 28},
+		{SENSOR_CHAN_ACCEL_Z, 2 << 28}, {SENSOR_CHAN_GYRO_X, 3 << 28},
+		{SENSOR_CHAN_GYRO_Y, 4 << 28},  {SENSOR_CHAN_GYRO_Z, 5 << 28},
+	};
+	static struct sensor_value scale;
+
+	/* 4g */
+	scale.val1 = 39;
+	scale.val2 = 226600;
+	zassert_ok(sensor_attr_set(dev, SENSOR_CHAN_ACCEL_XYZ, SENSOR_ATTR_FULL_SCALE, &scale));
+
+	/* 125 deg/s */
+	scale.val1 = 2;
+	scale.val2 = 181661;
+	zassert_ok(sensor_attr_set(dev, SENSOR_CHAN_GYRO_XYZ, SENSOR_ATTR_FULL_SCALE, &scale));
+
+	for (int i = 0; i < ARRAY_SIZE(values); ++i) {
+		zassert_ok(emul_sensor_backend_set_channel(accel_emul, values[i].channel,
+							   &values[i].value, 3));
+	}
 }
 
 static void *sensor_accel_setup(void)
 {
 	static struct sensor_accel_fixture fixture = {
 		.accel_spi = DEVICE_DT_GET(DT_ALIAS(accel_0)),
 		.accel_i2c = DEVICE_DT_GET_OR_NULL(DT_ALIAS(accel_1)),
+		.accel_emul_spi = EMUL_DT_GET(DT_ALIAS(accel_0)),
+		.accel_emul_i2c = EMUL_DT_GET_OR_NULL(DT_ALIAS(accel_1)),
 	};
 
+	sensor_accel_setup_emulator(fixture.accel_spi, fixture.accel_emul_spi);
+	sensor_accel_setup_emulator(fixture.accel_i2c, fixture.accel_emul_i2c);
+
```

Comment:
Is this change in the wrong commit? It doesn't seem related to `bmi160: Make changes to align driver to datasheet`

Reply:
This is correct, the extra tests here add coverage to the code that was fixed.